binary_transparency/firmware/common/signature.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2020 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha512"
+	"crypto/x509"
+	"encoding/pem"
+	"io/ioutil"
+	"strings"
+
+	"github.com/golang/glog"
+)
+
+func getPrivateKey() (*rsa.PrivateKey, error) {
+	mKey := strings.NewReader(RSApri)
+	priv, err := ioutil.ReadAll(mKey)
+	if err != nil {
+		glog.Exitf("Read failed! %s", err)
+	}
```

Comment:
This looks like it's taking a string, turning it into a reader, then reading all the bytes and turning them back into a string?

I think you can just delete these lines and do `pem.Decode(RSApri)` below?

Reply:
Had tried, it earlier, had issues hence the code is done this way! Decoding failed subsequently!